### DISCUSSION TITLE
Async API testing with "application/xml" contentType

### DISCUSSION THREAD

**Discussion by rakulkar-tibco** [microcks/microcks#601] [Q&A]
Hi,

How to handle async api testing (kafka) with content type as application/XML?How to add xml example ?Does anyone has an example for same?

**Comment by lbroudoux** [comment]
Hi @rakulkar-tibco

Thanks for opening this discussion. As of today, Microcks can publish mock messages with `application/xml` but testing and validating messages only manage `json` or `avro` content-types. Though as we already manage external schemas references (like we do with Avro) and XML validation with XSD, it shouldn't be hard to add this XML validation use-case.

Could you confirm my understanding of your needs is correct?

**Comment by rakulkar-tibco** [comment]
Yes, that is correct.
Could you please point to any document which explains publishing application/xml mock messages?

**Reply by lbroudoux** [reply]
Hum... do not have an example at hand but you simply have to set the contentType as message being application/xml in your AsyncAPI spec and then just specify Xml sample as a multiline string in your Yaml spec. 

Will try to produce a sample soon.
